java/src/main/java/ai/rapids/cudf/ColumnWriterOptions.java

```diff
@@ -537,6 +540,30 @@ public static ColumnWriterOptions mapColumn(String name, ColumnWriterOptions key
     return opt;
   }
 
+  /**
+   * Add a Map Column to the schema.
+   * <p>
+   * Maps are List columns with a Struct named 'key_value' with a child named 'key' and a child
+   * named 'value'. The caller of this method doesn't need to worry about this as this method will
+   * take care of this without the knowledge of the caller.
+   * @param isNullable is the returned map nullable.
+   * Note: If this map column is a key of another map, should pass isNullable = false.
+   * e.g.: map1(map2(int, int), int) the map2 should be non-nullable.
+   */
+  public static ColumnWriterOptions mapColumn(String name, ColumnWriterOptions key,
+                                              ColumnWriterOptions value, Boolean isNullable) {
+    StructColumnWriterOptions struct = structBuilder("key_value").build();
+    if (key.isNullable) {
+      throw new IllegalArgumentException("key column can not be nullable");
+    }
+    struct.childColumnOptions = new ColumnWriterOptions[]{key, value};
+    ColumnWriterOptions opt = listBuilder(name, isNullable)
```

Comment:
I am confused why we need a new API for this? Can we not just update the old API? It looks like it was a missed requirement and making something not support nulls should be fine for maps.

Reply:
Sorry forgot to delete this one when I understood the issue better.